src/app/panels/terminal.js

```diff
@@ -0,0 +1,420 @@
+    function mousedown (event) {
+      event.preventDefault()
+      if (event.which === 1) {
+        moveGhostbar(event)
+        document.body.appendChild(ghostbar)
+        document.addEventListener('mousemove', moveGhostbar)
+        document.addEventListener('mouseup', removeGhostbar)
+        document.addEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function cancelGhostbar (event) {
+      if (event.keyCode === 27) {
+        document.body.removeChild(ghostbar)
+        document.removeEventListener('mousemove', moveGhostbar)
+        document.removeEventListener('mouseup', removeGhostbar)
+        document.removeEventListener('keydown', cancelGhostbar)
+      }
+    }
+    function moveGhostbar (event) { // @NOTE HORIZONTAL ghostbar
+      ghostbar.style.top = self._api.getPosition(event) + 'px'
+    }
+    function removeGhostbar (event) {
+      if (self._view.icon.classList.contains('fa-angle-double-up')) {
+        self._view.icon.classList.toggle('fa-angle-double-down')
+        self._view.icon.classList.toggle('fa-angle-double-up')
+      }
+      document.body.removeChild(ghostbar)
+      document.removeEventListener('mousemove', moveGhostbar)
+      document.removeEventListener('mouseup', removeGhostbar)
+      document.removeEventListener('keydown', cancelGhostbar)
+      self.event.trigger('resize', [self._api.getPosition(event)])
+    }
+
+    return self._view.panel
+
+    function change (event) {
+      if (event.which === 13) {
+        if (event.ctrlKey) { // <ctrl+enter>
+          self._view.input.appendChild(document.createElement('br'))
+          self.scroll2bottom()
+          putCursor2End(self._view.input)
+        } else { // <enter>
+          event.preventDefault()
+          self.execute(self._view.input.innerText)
+          self._view.input.innerHTML = ''
+        }
+      }
+    }
+    function putCursor2End (editable) {
+      var range = document.createRange()
+      range.selectNode(editable)
+      var child = editable
+      var chars
+
+      while (child) {
+        if (child.lastChild) child = child.lastChild
+        else break
+        if (child.nodeType === Node.TEXT_NODE) {
+          chars = child.textContent.length
+        } else {
+          chars = child.innerHTML.length
+        }
+      }
+
+      range.setEnd(child, chars)
+      var toStart = true
+      var toEnd = !toStart
+      range.collapse(toEnd)
+
+      var sel = window.getSelection()
+      sel.removeAllRanges()
+      sel.addRange(range)
+
+      editable.focus()
+    }
+  }
+  _log (mode) {
+    var self = this
+    var modes = { log: true, info: true, error: true }
+    if (modes[mode]) {
+      return function logger () {
+        var args = [].slice.call(arguments)
+        self.data.session.push(args)
+        var types = args.map(type)
+        var values = javascriptserialize.apply(null, args).map(function (x, i) {
+          return (typeof args[i] === 'string') ? args[i] : x
```

Comment:
hmm, interesting. yes I think you might be right xD wow... good catch. you crazy guy actually read and try to understand the code? :P haha

Was I drunk? I actually have no clue what this should do. It makes no sense to me now.
Like `typeof args[i] === 'string'` should always be **true** because that's the whole point of `javascriptserialize` :p 

I copied it from my module and adapted it, see here: https://github.com/serapath/dom-console/blob/master/domconsole.js#L195

Thx, for spotting it, i'll just remove it.